subsystems/networking/networkmanager.go

```diff
@@ -172,23 +175,72 @@ func (n *Networking) StartProvisioning(ctx context.Context, inputChan chan<- use
 	n.opMu.Lock()
 	defer n.opMu.Unlock()
 
-	n.logger.Info("Starting provisioning mode.")
-	_, err := n.addOrUpdateConnection(utils.NetworkDefinition{
-		Type:      NetworkTypeHotspot,
-		Interface: n.Config().HotspotInterface,
-		SSID:      n.Config().HotspotSSID,
-	})
-	if err != nil {
-		return err
-	}
-	if err := n.activateConnection(ctx, n.Config().HotspotInterface, n.Config().HotspotSSID); err != nil {
-		return errw.Wrap(err, "starting provisioning mode hotspot")
-	}
+	n.logger.Info("Starting provisioning mode. Attempting to set up both hotspot and bluetooth provisioning methods...")
 
-	// start portal with ssid list and known connections
-	if err := n.startPortal(inputChan); err != nil {
-		err = errors.Join(err, n.deactivateConnection(n.Config().HotspotInterface, n.Config().HotspotSSID))
-		return errw.Wrap(err, "starting web/grpc portal")
+	// Simultaneously start both the hotspot captive portal and the bluetooth service provisioning methods.
+	wg := sync.WaitGroup{}
+
+	wg.Add(1)
+	var hotspotErr error
+	goutils.ManagedGo(func() {
+		_, err := n.addOrUpdateConnection(utils.NetworkDefinition{
+			Type:      NetworkTypeHotspot,
+			Interface: n.Config().HotspotInterface,
+			SSID:      n.Config().HotspotSSID,
+		})
+		if err != nil {
+			hotspotErr = err
+			return
+		}
+		if err := n.activateConnection(ctx, n.Config().HotspotInterface, n.Config().HotspotSSID); err != nil {
+			hotspotErr = errw.Wrap(err, "starting provisioning mode hotspot")
+			return
+		}
+
+		// start portal with ssid list and known connections
+		if err := n.startPortal(inputChan); err != nil {
+			err = errors.Join(err, n.deactivateConnection(n.Config().HotspotInterface, n.Config().HotspotSSID))
+			hotspotErr = errw.Wrap(err, "starting web/grpc portal")
+			return
+		}
+		n.hotspotIsActive = true
+	}, wg.Done)
+
+	wg.Add(1)
+	var bluetoothErr error
+	goutils.ManagedGo(func() {
+		if n.bluetoothService == nil {
+			bt, err := newBluetoothService(
+				n.logger, fmt.Sprintf("%s.%s.%s", n.cfg.Manufacturer, n.cfg.Model, n.cfg.FragmentID), n.getVisibleNetworks())
```

Comment:
the config values should be n.Config() otherwise you can hit a race.
```suggestion
				n.logger, fmt.Sprintf("%s.%s.%s", n.Config().Manufacturer, n.Config().Model, n.Config().FragmentID), n.getVisibleNetworks())
```

Reply:
n.getVisibleNetworks() is a one-time call here, but the list of visible networks changes frequently, especially if someone is trying to get a new network to show up. This needs to be handled like portal does, and updated when things change, not passed in once here only at startup.

Reply:
1. `n.Config()` - will use thank you for catching, silly mistake.
2. `n.getVisibleNetworks()` - I initially tried to have a channel to update the current value stored in the "available wifi network" characteristic, but this didn't work properly. I need to look into whether it is easily possible to tear down an active characteristic and post a new one while the BT service is already being advertised. The issue here is I don't want to tear down the BT service so that I can update available WiFi networks while the client could be inputting their Robot Part ID (i.e. trying to input information that has nothing to do with the WiFi networks)